### ISSUE TITLE
.Net: PromptExecutionSettings in KernelArguments should be an ordered dictionary

### ISSUE THREAD

**Issue by matthewbolanos** [microsoft/semantic-kernel#4212]
tl;dr, it's not possible to pass multiple execution settings (one for each model ID, service ID, or another key) into an `IAIServiceSelector`

I have the following prompt

```csharp
var chat = kernel.CreateFunctionFromPrompt(
    new PromptTemplateConfig()
    {
        Name = "Chat",
        Description = "Chat with the assistant.",
        Template = @"{{ConversationSummaryPlugin.SummarizeConversation $history}}
        User: {{$request}}
        Assistant: ",
        TemplateFormat = "semantic-kernel",
        InputVariables = [
            new() { Name = "history", Description = "The history of the conversation.", IsRequired = false, Default = "" },
            new() { Name = "request", Description = "The user's request.", IsRequired = true }
        ],
        ExecutionSettings = [
            new() {
                // Default settings for all models
                ExtensionData = new() {
                    { "MaxTokens", 1000 },
                    { "Temperature", 0 }
                }
            },
            new() {
                // Settings for gpt-3.5-turbo
                ModelId = "gpt-3.5-turbo",
                ExtensionData = new() {
                    { "MaxTokens", 4000 },
                    { "Temperature", 0.1 }
                }
            },
            new() {
                // Settings for gpt-4
                ModelId = "gpt-4-1106-preview",
                ExtensionData = new() {
                    { "MaxTokens", 8000 },
                    { "Temperature", 0.3 }
                }
            }
        ]
    }
);
```

Notice that I have settings to support 3 different models. During invocation time, however, I want to override these settings using `KernelArguments`. Unfortunately, `KernelArguments` only lets me define a single `PromptExecutionSetting` and at this point, I don't know which model is best because that's the responsibility of the `IAIServiceSelector`

```csharp
var result = await kernel.InvokeAsync<string>(chat, new(new PromptExecutionSettings()
{
    ModelId = "gpt-3.5-turbo",
    ExtensionData = new() { { "MaxTokens", 1000 } }
}) { { "input", "Hello world!" } });
```

Once I'm in the `IAIServiceSelector`, I only get a single `PromptExecutionSettings`, so if the service selector chooses a different model than I provided in the `KernelArguments`, then I won't be able to override it corectly

```csharp
public bool TrySelectAIService<T>(
        Kernel kernel, KernelFunction function, KernelArguments arguments,
        [NotNullWhen(true)] out T? service, out PromptExecutionSettings? serviceSettings) where T : class, IAIService
```

**Comment by strikene** [comment]
I have four models with GPT4 models, in this case, what should I do?
